x-pack/legacy/plugins/reporting/export_types/printable_pdf/server/execute_job/index.ts

```diff
@@ -67,7 +73,36 @@ export const executeJobFactory: QueuedPdfExecutorFactory = async function execut
       })
     );
 
-    const stop$ = Rx.fromEventPattern(cancellationToken.on);
-    return process$.pipe(takeUntil(stop$)).toPromise();
+    const timeout$ = Rx.fromEventPattern(
```

Comment:
Not sure if Rx has a more idiomatic pattern for doing setTimeout's, but this was the best I could come up (happy to change it if there's a better operator).

Reply:
Maybe [delay](https://rxjs-dev.firebaseapp.com/api/operators/delay) ?

Reply:
How about:
```
    const timeout$: Rx.Observable<JobDocOutput> = Rx.timer(timeout).pipe(
      mergeMap(() => {
        jobLogger.warn('Screenshot timeout breach, attempting to capture page screenshot');
        return browserDriverFactory.screenshotPage().then(lastScreen => ({
          content_type: 'image/png',
          content: lastScreen.toString('base64'),
          size: lastScreen.byteLength,
          warnings: ['Screenshot timed out, debug screenshot captured'],
        }));
      }),
      catchError(err => {
        jobLogger.error(err);
        return Rx.throwError(err);
      })
    );
```